test/node/features/delivery.feature

```diff
@@ -0,0 +1,21 @@
+Feature: Delivery of errors
+
+Background:
+  Given I store the api key in the environment variable "BUGSNAG_API_KEY"
+  And I store the notify endpoint in the environment variable "BUGSNAG_NOTIFY_ENDPOINT"
+  And I store the sessions endpoint in the environment variable "BUGSNAG_SESSIONS_ENDPOINT"
+  And I store the logs endpoint in the environment variable "BUGSNAG_LOGS_ENDPOINT"
+
+Scenario: Delivery for an oversized error is not retried
+  Given I start the service "express"
+  And I wait for the host "express" to open port "80"
+  And I set the HTTP status code for the next "POST" request to 400
+  When I open the URL "http://express/oversized"
+  And I wait for 5 seconds
+  Then I wait to receive an error
+
+  # Check that Bugsnag is discarding the event
+  And I wait to receive 3 logs
+  Then I discard the oldest log
+  Then I discard the oldest log
+  And the log payload field "message" equals "Event oversized (5.02 MB)"
```

Comment:
@Cawllec whilst persistence is not enabled in node yet I'd be inclined (if it's easy) to add further steps to assert when the app is reloaded no sends are attempted. Is that currently possible?

Reply:
Absolutely, although you may need to add a step locally that calls `Maze::Docker.down_all_services`, something like:
```
Then('I stop all docker services') do
    Maze::Docker.down_all_services
end
```
and then empty out the old stack of payloads and restart the service:
```
Then I stop all docker services
And I discard the oldest error
And I start the service "express"
And I wait for the host "express" to open port "80"
And I should receive 0 requests
```